flowmachine/flowmachine/core/query.py

```diff
@@ -830,64 +830,75 @@ def invalidate_db_cache(self, name=None, schema=None, cascade=True, drop=True):
-
-                if cascade:
-                    for dep_id in dep_ids:
-                        dep = get_obj_or_stub(get_db(), dep_id)
-                        log(
-                            "Cascading to dependent.",
-                            dependency=dep.fully_qualified_table_name,
+                    ]
+                    with con.begin():
+                        con.execute(
+                            "DELETE FROM cache.cached WHERE query_id=%s",
+                            (self.query_id,),
                         )
-                        dep.invalidate_db_cache()
+                        log("Deleted cache record.")
+                        if drop:
+                            con.execute(
+                                "DROP TABLE IF EXISTS {}".format(
+                                    self.fully_qualified_table_name
+                                )
+                            )
+                            log("Dropped cache table.")
+
+                    if cascade:
+                        for dep_id in dep_ids:
+                            dep = get_obj_or_stub(get_db(), dep_id)
+                            log(
+                                "Cascading to dependent.",
+                                dependency=dep.fully_qualified_table_name,
+                            )
+                            dep.invalidate_db_cache()
+                    else:
+                        log("Not cascading to dependents.")
+                except NotImplementedError:
+                    logger.info("Table has no standard name.")
+                # Outside of cache schema table
+                if schema is not None:
+                    full_name = "{}.{}".format(schema, name)
+                    log("Dropping table outside cache schema.", table_name=full_name)
                 else:
-                    log("Not cascading to dependents.")
-            except NotImplementedError:
-                logger.info("Table has no standard name.")
-            # Outside of cache schema table
-            if schema is not None:
-                full_name = "{}.{}".format(schema, name)
-            else:
-                full_name = name
-            log("Dropping table outside cache schema.", table_name=full_name)
-            with con.begin():
-                con.execute("DROP TABLE IF EXISTS {}".format(full_name))
-            q_state_machine.finish_resetting()
+                    full_name = name
+                with con.begin():
+                    con.execute("DROP TABLE IF EXISTS {}".format(full_name))
+                q_state_machine.finish_resetting()
+            except Exception as exc:
+                logger.error(
+                    "Query reset failed.",
+                    query_id=self.query_id,
+                    action="invalidate_db_cache",
+                    exception=exc,
+                )
+                q_state_machine.raise_error()
+                raise exc
```

Comment:
Is there a potential for hang here, if a query can reset from elsewhere? This thread throws Exception, some cache service sees it and tries again, this thrown Exception, ect....

Reply:
Yes, something to be wary of - should have enough information to avoid that because we arrive in an explicit reset failed state.